### PR TITLE
feat: add indexer test code

### PR DESCRIPTION
**Pull request by bitrocks**
## database setup
The mocked database for testing is based on the first 100 blocks from Lina mainnet,  [blocks_data.json](https://github.com/nervosnetwork/lumos/blob/indexer-tests/packages/indexer/tests/blocks_data.json). 

## test cases
- [x] CellCollector query cases
- [x] TransactionCollector query cases
- [x] Indexer.subscribe query cases

The same blocks_data and test cases should be tested in both @ckb-lumos/indexer and @ckb-lumos/sql-indexer.

### CODE REVIEW

File: packages/indexer/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
+{
+  "name": "@ckb-lumos/indexer",
```
Comment (classicalliu) [inline packages/indexer/package-lock.json]: Why this file exists ?

File: packages/indexer/tests/cell_collector.test.js [hunk 2]
```diff
@@ -1,40 +1,44 @@
 const test = require("ava");
 const { CellCollector } = require("../lib");
+const { Indexer } = require("./helper.js");
+const { lock, type, cellCollectorTestCases } = require("./test_cases.js");
+// the node_uri will not be connected during the test process, only serves as a placeholder when create an indexer instance.
+const node_uri = "http://127.0.0.1:8114";
```
Comment (classicalliu) [inline packages/indexer/tests/cell_collector.test.js]: `nodeUri` may better.

File: packages/indexer/tests/indexer.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const test = require("ava");
+const sinon = require("sinon");
+const fs = require("fs");
+const { Indexer } = require("./helper.js");
+const { lock, type, indexerSubscribeTestCases } = require("./test_cases.js");
+// the node_uri will not be connected during the test process, only serves as a placeholder when create an indexer instance.
+const node_uri = "http://127.0.0.1:8116";
```
Comment (classicalliu) [inline packages/indexer/tests/indexer.test.js]:
```suggestion
const nodeUri = "http://127.0.0.1:8116";
```

File: packages/indexer/tests/indexer.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+const tmpIndexedDataPath = "/tmp/indexed_data3";
+const blocksDataFilePath = __dirname + "/blocks_data.json";
+const indexer = new Indexer(node_uri, tmpIndexedDataPath);
```
Comment (classicalliu) [inline packages/indexer/tests/indexer.test.js]:
```suggestion
const indexer = new Indexer(nodeUri, tmpIndexedDataPath);
```

File: packages/indexer/tests/transaction_collector.test.js [hunk 5]
```diff
@@ -1,43 +1,52 @@
...
 const { TransactionCollector } = require("../lib");
+const { Indexer } = require("./helper.js");
+const {
+  lock,
+  type,
+  transactionCollectorTestCases,
+} = require("./test_cases.js");
+const fs = require("fs");
+// the node_uri will not be connected during the test process, only serves as a placeholder when create an indexer instance.
+const node_uri = "http://127.0.0.1:8115";
```
Comment (classicalliu) [inline packages/indexer/tests/transaction_collector.test.js]:
```suggestion
const nodeUri = "http://127.0.0.1:8115";
```

File: packages/indexer/tests/transaction_collector.test.js [hunk 6]
```diff
@@ -1,43 +1,52 @@
...
+const tmpIndexedDataPath = "/tmp/indexed_data2";
+const blocksDataFilePath = __dirname + "/blocks_data.json";
+const indexer = new Indexer(node_uri, tmpIndexedDataPath);
```
Comment (classicalliu) [inline packages/indexer/tests/transaction_collector.test.js]:
```suggestion
const indexer = new Indexer(nodeUri, tmpIndexedDataPath);
```

File: packages/indexer/package-lock.json (hunk 1 above)
Comment (bitrocks) [reply packages/indexer/package-lock.json]: Oops... removed.

**Review by classicalliu — APPROVED** [review]

**Pull request merged**